src/string.cr

```diff
@@ -499,33 +529,28 @@ class String
     table
   end
 
-  # :nodoc:
-  record ToU64Info,
-    value : UInt64,
-    negative : Bool,
-    invalid : Bool
```

Comment:
This can parameterize over the integer type directly:

```crystal
record ToUnsignedInfo(T), value : T, # ...
```

There is no need to switch to a `NamedTuple`.

Reply:
I have improved this in #11093